app/services/fact_importer.rb

```diff
@@ -111,7 +115,11 @@ def normalize(facts)
     facts.keep_if { |k, v| v.present? }
   end
 
+  def db_facts_as_relation
+    host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'")
+  end
+
   def db_facts
-    @db_facts ||= host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'").index_by(&:name)
+    @db_facts ||= db_facts_as_relation.index_by(&:name)
```

Comment:
Why do we need the hash? we only use the hash keys in the `add_new_facts` method, is there anywhere we use the hash values after your change?